compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -182,6 +183,15 @@ pub struct NodeInterner {
     /// and creating a `Token::QuotedType(id)` from this id. We cannot create a token holding
     /// the actual type since types do not implement Send or Sync.
     quoted_types: noirc_arena::Arena<Type>,
+
+    /// Store the location of the references in the graph
+    pub(crate) reference_graph: DiGraph<DependencyId, ()>,
+
+    /// Tracks the index of the references in the graph
+    pub(crate) reference_graph_indices: HashMap<DependencyId, PetGraphIndex>,
+
+    /// Store the location of the references in the graph
+    pub(crate) location_indices: LocationIndices,
```

Comment:
💭 One thing I'm thinking is that these locations and references are now tracked in all compilations, even thought they are only needed by the LSP code. Ideally there's a boolean controlling not to track this, but the interner is created deep in the code so getting a boolean from `check_crate` up to that point is kind of tricky. It could be done in a separate PR, though, and only if really needed (if we notice this change is making compilation times slow).

Reply:
Yes, this is something I'm a little concerned about in the long-term. Having a clean separation between things which are necessary for the compiler to function and "optional" features would be really good.

Reply:
Not required for this PR however to be clear.